impl/src/main/java/org/jboss/weld/injection/ResourceInjectionProcessor.java

```diff
@@ -90,8 +90,7 @@ protected <T> Set<ResourceInjection<?>> createResourceInjections(Bean<?> declari
         }
 
         Class<? extends Annotation> marker = getMarkerAnnotation(processorContext);
-
-        final Collection<EnhancedAnnotatedField<?, ? super T>> fields = type.getDeclaredEnhancedFields(marker);
+        final Collection<EnhancedAnnotatedField<?, ? super T>> fields = type.getEnhancedFields(marker);
         final Collection<EnhancedAnnotatedMethod<?, ? super T>> methods = type.getDeclaredEnhancedMethods(marker);
```

Comment:
For methods, the `EnhancedAnnotatedType` seems to have different logic and "declared" methods contain all of the methods of the bean while trying to use `getEnhancedMethods(marker)` fails as that is a map containing only entries for a subset of annotations.

I am not sure if that's an optimization or an oversight; I will keep poking around that.
Nonetheless, the fix in this PR is sufficient to fix the problem in the linked issue.